### PR TITLE
[CHK-2828] Check user in transactions operations

### PR DESCRIPTION
**Pull request by ciuffagianluca**
Add check users for the following operations:

- GET transaction
- DELETE transaction
- POST authorization-request

#### List of Changes

APIs definition accept an header field `x-user-id` used to be checked against the transaction document user id value. This check is applied only for v2 document and when the userId is set. 
So we can distinguish 2 case and relative sub-cases:

1.  `V1` document: check is done against the value of the xUserId passed from the query. If it is null the check is ok. Otherwise (not accepted case by design, `V1` document hasn't the field `userId` ) the result is cut.
2. `V2` document: check is done against the value of the `xUserId` passed from the query:
     - if it is null and the document field is null, the result is accepted
     - if is in not null, the check is done from xUserId.toString() against document.userId field. If the values are equal. the result is accepted.      

If the field is null and the transaction is from v2 domains, it will be checked the null value for userId in the document.

#### Motivation and Context

Add security check for authenticated transactions (appIO)

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/transactions/repositories/TransactionsViewRepository.java [hunk 1]
```diff
@@ -1,9 +1,11 @@
 package it.pagopa.transactions.repositories;
 
 import it.pagopa.ecommerce.commons.documents.BaseTransactionView;
+import org.springframework.data.mongodb.repository.Query;
 import org.springframework.data.repository.reactive.ReactiveCrudRepository;
 import reactor.core.publisher.Mono;
 
 public interface TransactionsViewRepository extends ReactiveCrudRepository<BaseTransactionView, String> {
     Mono<BaseTransactionView> findByTransactionId(String transactionId);
+
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/repositories/TransactionsViewRepository.java:10]: Stray empty line
```suggestion
```

File: src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java [hunk 2]
```diff
@@ -711,6 +712,15 @@ public Mono<RequestAuthorizationResponseDto> requestTransactionAuthorization(
                 );
     }
 
+    private Mono<BaseTransactionView> getBaseTransactionView(String transactionId, UUID xUserId) {
+        return transactionsViewRepository.findById(transactionId)
+                    .filter(transactionDocument -> switch (transactionDocument) {
+                        case it.pagopa.ecommerce.commons.documents.v1.Transaction ignored -> true;
+                        case it.pagopa.ecommerce.commons.documents.v2.Transaction t -> xUserId == null ? t.getUserId() == null : t.getUserId().equals(xUserId.toString());
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java:719]: Alternative:
```suggestion
                        case it.pagopa.ecommerce.commons.documents.v2.Transaction t -> Objects.equals(xUserId.toString(), t.getUserId());
```

**Review by petretiandrea — DISMISSED** [review]

File: src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java [hunk 3]
```diff
@@ -711,6 +712,15 @@ public Mono<RequestAuthorizationResponseDto> requestTransactionAuthorization(
...
+                        default -> throw new NotImplementedException("Handling for transaction document: [%s] not implemented yet".formatted(transactionDocument.getClass()));
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java:720]:
```suggestion
                        default -> throw new NotImplementedException("Handling for transaction document version: [%s] not implemented yet".formatted(transactionDocument.getClass()));
```

File: src/test/java/it/pagopa/transactions/services/v2/TransactionServiceTests.java [hunk 4]
```diff
@@ -283,15 +283,85 @@ void getTransactionReturnsTransactionDataOriginProvided() {
...
 
-        StepVerifier.create(transactionsServiceV1.getTransactionInfo(TRANSACTION_ID))
+        StepVerifier
+                .create(transactionsServiceV1.getTransactionInfo(TRANSACTION_ID, UUID.fromString(USER_ID)))
                 .expectNext(expected)
                 .verifyComplete();
     }
 
+    @Test
+    void getTransactionReturnsTransactionV2() {
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/transactions/services/v2/TransactionServiceTests.java:297]:
```suggestion
    void getTransactionReturnsTransactionV2WithNullUserId() {
```

**Review by giovanniberti — APPROVED** [review]

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**